webknossos/webknossos/dataset/dataset.py

```diff
@@ -441,6 +505,178 @@ def download(
                 exist_ok=exist_ok,
             )
 
+    @classmethod
+    def from_images(
+        cls,
+        input_path: Union[str, PathLike],
+        output_path: Union[str, PathLike],
+        voxel_size: Optional[Tuple[float, float, float]] = None,
+        name: Optional[str] = None,
+        *,
+        map_filepath_to_layer_name: Union[
+            ConversionLayerMapping, Callable[[Path], str]
+        ] = ConversionLayerMapping.INSPECT_SINGLE_FILE,
+        z_slices_sort_key: Callable[[Path], Any] = natsort_keygen(),
+        enforce_category: Optional[LayerCategoryType] = None,
+        data_format: Union[str, DataFormat] = DEFAULT_DATA_FORMAT,
+        chunk_shape: Optional[Union[Vec3IntLike, int]] = None,
+        chunks_per_shard: Optional[Union[int, Vec3IntLike]] = None,
+        compress: bool = False,
```

Comment:
Is there a benefit to not compressing? I would have expected the default to be True

Reply:
Compression would normally happen afterwards in a separate step, as directly writing compressed chunks requires to write whole shards, which defaults to reading 1024 slices at once instead of 32. If we add a CLI, this would probably run `from_images, compress, downsample` once after the other, but I'd rather not add a separate compression step in this function. Does this make sense?

Reply:
In the latest commit 25d11cb22564e8f3639c00b3f7ae9b68ff5d8287 I added post-hoc compression and downsampling, defaulting to true for both. I also added a small logic to compress directly when converting if the number of slices is small (`<= z-chunk-size`). I'm not sure if that's over-engineered, tbh., and if that actually saves space (I think so, though).
cc @normanrz, What do you think about this?

Reply:
Hmm. I'd err on the side of simplicity: Keep post-compression and downsample out of the function and leave the old compress option in. For convenience, we could add `Dataset.compress` and `Dataset.downsample` to compress and downsample all layers/mags.

Reply:
Sounds good! I reverted those changes in this PR, `ds.compress` and `ds.downsample` will come in another PR.